gtfs/spec/en/reference.md

```diff
@@ -612,6 +624,69 @@ Describes levels in a station. Useful in conjunction with `pathways.txt`, and is
 |  `level_index` | Float | **Required** | Numeric index of the level that indicates its relative position. <br><br>Ground level should have index `0`, with levels above ground indicated by positive indices and levels below ground by negative indices.|
 |  `level_name` | Text | Optional | Name of the level as seen by the rider inside the building or station.<hr>_Example: Take the elevator to "Mezzanine" or "Platform" or "-1"._|
 
+### location_groups.txt
+
+File: **Optional**
+
+Primary key (`location_group_id`, `location_id`)
+
+Defines location groups that indicate groups of stops where a rider may request pickup or drop off.
+
+| Field Name | Type | Required | Description |
+| ---------- | ---- | ------------ | ----------- |
+| `location_group_id` | ID | **Required** | Identifies a location group. ID must be unique across all `stops.stop_id`, locations.geojson `id`, and `location_groups.location_group_id` values. <br><br>A location group is a group of stops that together indicate locations where a rider may request pickup or drop off.<br><br>Multiple entries in `location_groups.txt` can have the same `location_group_id`. | 
+| `stop_id` | Foreign ID referencing `stops.stop_id` | **Required** | Identifies a stop belonging to the location group. |
+| `location_group_name` | Text | Optional | Name of the location group. Must be defined either once, or exhaustively for a single `location_group_id`. |
+
+### locations.geojson
+
+File: **Optional**
+
+Defines GeoJSON locations that indicate groups of lat/lon coordinates defining zones where riders can request either pickup or drop off.
```

Comment:
Wording here could also be confusing. The definition feels a bit circular: "Locations... defines... locations that indicate groups." Including more of the sentence, this can be condensed to "Locations defines groups... defining zones", which can be further condensed to "Locations define zones".  Is it then fair to say that these locations are just zones? I realize terms stick around for historical reasons, but at this point in the evolution of the spec, would it make more sense to just call these "zones" or "flex_zones"? 

I think there's also the consistency issue I mentioned in another comment, where currently location_groups are not in fact groups of locations. 

Do "groups of lat/lon coordiantes" refer to the interior of the polygons, or to the points defining the perimeter of the polygon? In either case these seem to be details of how the zones are represented or defined, and not the nature of the entities themselves. I believe they're just zone polygons.

```suggestion
Defines zones where riders can request either pickup or drop off by flex services. These zones are represented as GeoJSON polygons.
```

Reply:
Agree with the suggested wording change for line 645.

Reply:
The above comment https://github.com/google/transit/pull/388#discussion_r1417750416 also applies here: the term "flex" doesn't seem to be used in the spec, so some other expression like "on-demand" might be used. 

However, "flex" is clearly a distinct idea in the minds of people drafting the spec, which seems to be intended as a single coherent extension to the spec, and introduces new entities that are specific to this extension and this kind of transportation. It introduces elements that many people will never use, and other people will need to focus on as a specific domain in data pipelines or routing. Both groups need to be able to mentally categorize the growing number of entity types, understand readily which tables "go together" and which ones they need to examine or ignore depending on their use case. To meet those needs, it seems reasonable to qualify the names of new entities with the extension or specific kind of transportation they relate to. This could be done with a prefix, preferably a short memorable one that matches the term used in the proposal or other documents. `flex_` seems like a pretty good candidate to me.

My main point here though was that these may only called "locations" for historical reasons, and it may have long-term benefit to settle on names that better reflect the current role of the entities before the extension is formally adopted. Chains of definitions were built up to say that "locations" are actually geographic polygons that people call "zones" in this kind of transit. In that case, is there any reason they aren't just called "zones"? And if "zones" might be too general because there are also things like "fare zones", then aren't these just on-demand zones or "flex_zones"? That's a nice, short name that says exactly what the table contains and clearly groups it with other tables and concepts only used in on-demand or flex routing, and with proposals and documents pertaining to on-demand or flex routing.

Reply:
I’m hearing you on a lot of your points @abyrd. The volume of commentary here makes it difficult to respond in writing comprehensively, and doing so would only make it harder for other readers to track discussion. As such, I suggest we discuss these topics at the next Flex meeting. @tzujenchanmbd 

I see two main issues you’ve raised:
1. Should we rework location_groups.txt to include normalization of location_groups in a separate file (similar structure to areas/stop_areas)?
  e.g., location_groups.txt + location_group_stops.txt (assuming naming convention remains the same)
2. Should we rework naming convention of location_groups.txt and maybe locations.geojson and use something like “flex_groups.txt (+ flex_group_stops.txt)” and “flex_zones.geojson”?

Does that sound right?

Reply:
To that I would add:
a) Rather than just prefixing extension names, consider completely renaming entities to narrow meaning and resolve contradictions (e.g. location_groups are not made of locations) (this may already be implied in your point 2)
b) Extend table naming scheme uniformly to other proposals and tables, e.g.: `extension_entity[_subentity]`
c) Consider introducing additional entities or "superclasses" to the specification that coherently capture why different tables share an ID namespace. e.g. flex routes reference a stop-like entity class that encompass normal stops, but also a couple of other things. These do not need to be materialized in tables, but their existence in the specification and description could make it much easier to understand.

Note that I will probably not be able to attend any meetings on this topic. But please feel free to take my commentary into consideration when working on next steps at any planned meeting.  

I think more so than written vs. spoken communication, there are other aspects of the change proposal and adoption process that impede clear understanding and discussion. But I hope to raise those points on a new issue or #413, not on this pre-existing proposal.

Reply:
I just added #415 which I think relates to the discussion above.

Reply:
Just quickly answer some questions here -
> Just to be sure: the values appearing in areas.area_id and those in stop_areas.area_id are drawn from exactly the same set of values right? The set is just defined by the unique values in areas, then those same values are referenced in stop_areas? In standard terminology, they are the primary keys of areas and foreign keys in stop_areas?

Yes.

> The volume of commentary here makes it difficult to respond in writing comprehensively, and doing so would only make it harder for other readers to track discussion. As such, I suggest we discuss these topics at the next Flex meeting.

We are planning a meeting, and will announce more details (such as the date and link) shortly.

Reply:
Modified locations.geojson file description - https://github.com/google/transit/pull/388/commits/1419292118398bf9388f707e882d468a30839d6e

Field name discussed in the 1/17 meeting, resolving this conversation.